Kitodo-DataAccess/src/test/java/org/kitodo/dataaccess/storage/memory/MemoryNodeReferenceTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.dataaccess.storage.memory;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.fail;
+
+import org.junit.Test;
+import org.kitodo.dataaccess.storage.memory.MemoryNodeReference;
+
+public class MemoryNodeReferenceTest {
+
+    @Test
+    public void testEqualsIsFalseForDifferentObjects() {
+        MemoryNodeReference one = new MemoryNodeReference("http://names.example/foo");
+        MemoryNodeReference other = new MemoryNodeReference("http://names.example/bar");
+
+        assertThat(one, is(not(equalTo(other))));
+    }
+
+    @Test
+    public void testEqualsObject() {
+        MemoryNodeReference one = new MemoryNodeReference("http://names.example/foo");
+        MemoryNodeReference another = new MemoryNodeReference("http://names.example/foo");
+
+        assertThat(one, is(equalTo(another)));
+    }
+
+    @Test
+    public void testGetIdentifier() {
+        assertThat(new MemoryNodeReference("http://names.example/foo").getIdentifier(),
+            is(equalTo("http://names.example/foo")));
+    }
+
+    @Test
+    public void testHashCodeDiffersForTwoDifferentInstances() {
+        MemoryNodeReference one = new MemoryNodeReference("http://names.example/foo");
+        MemoryNodeReference other = new MemoryNodeReference("http://names.example/bar");
+
+        assertThat(one.hashCode() == other.hashCode(), is(false));
+    }
+
+    @Test
+    public void testHashCodeIsEqualForTwoEqualInstances() {
+        MemoryNodeReference one = new MemoryNodeReference("http://names.example/foo");
+        MemoryNodeReference other = new MemoryNodeReference("http://names.example/foo");
+
+        assertThat(one.hashCode() == other.hashCode(), is(true));
+    }
+
+    @Test
+    public void testMemoryNodeReferenceCanBeCreatedFromHttpUrl() {
+        new MemoryNodeReference("http://names.example/foo#bar");
+    }
+
+    @Test
+    public void testMemoryNodeReferenceCanBeCreatedFromNonHttpUrl() {
+        new MemoryNodeReference("urn:example:kitodo-123456789-xyz");
```

Comment:
What did you testing here?

Reply:
Tests that URNs are accepted as an URI, no exception being thrown.

Reply:
https://stackoverflow.com/questions/17731234/how-to-test-that-no-exception-is-thrown

Reply:
@Beacze , see comment to question what you posted: “A JUnit test is judged to have failed if it throws any exception other than an expected exception. Usually no exceptions are expected.” So this is perfectly fine.

Reply:
Actually I was more meaning that we could use for .e.g assertNull and check if the exception is null, as test case which is here is very misleading.

Reply:
What you mean would be like this:

    Throwable t = null;
    try {
        new MemoryNodeReference("urn:example:kitodo-123456789-xyz");
    } catch (Throwable t1) {
        t = t1;
    }
    assertThat(t, is(nullValue()));
But actually this is overkill. It is very OK to have tests without exceptions if the only condition of the test is that a method does not throw an exception. It is the default behaviour of the test framework.